haystack_experimental/dataclasses/tool.py

```diff
@@ -4,16 +4,27 @@
 
 import inspect
 from dataclasses import asdict, dataclass
-from typing import Any, Callable, Dict
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Union
 
 from haystack.lazy_imports import LazyImport
+from haystack.logging import logging
 from haystack.utils import deserialize_callable, serialize_callable
 from pydantic import create_model
 
+from haystack_experimental.dataclasses.types import OpenAPIKwargs
+
+logger = logging.getLogger(__name__)
+
 with LazyImport(message="Run 'pip install jsonschema'") as jsonschema_import:
     from jsonschema import Draft202012Validator
     from jsonschema.exceptions import SchemaError
 
+with LazyImport(message="Run 'pip install openapi-llm'") as openapi_llm_import:
+    from openapi_llm.client.config import ClientConfig
+    from openapi_llm.client.openapi import OpenAPIClient
+    from openapi_llm.core.spec import OpenAPISpecification
```

Comment:
In general, I see that offloading most of the logic in Vladimir's library results for us in less maintenance effort (and I am happy about that).
Nevertheless, I see some risks. @julian-risch what's your take on this aspect?